clusterctl/cmd/validate_cluster.go

```diff
@@ -17,26 +17,130 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
+
 	"github.com/golang/glog"
 	"github.com/spf13/cobra"
-	"sigs.k8s.io/cluster-api/errors"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
 )
 
+type ValidateClusterOptions struct {
+	ClusterName string
+	Namespace   string
+}
+
+var vco = &ValidateClusterOptions{}
+
 var validateClusterCmd = &cobra.Command{
 	Use:   "cluster",
 	Short: "Validate a cluster created by cluster API.",
 	Long:  `Validate a cluster created by cluster API.`,
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := RunValidateCluster(); err != nil {
-			glog.Exit(err)
-		}
+		RunValidateCluster()
```

Comment:
I think i would keep the single glog.Exit (...) call to match the other commands in the application and make the application simpler to understand for contributing developers. If we are to change that, we should change it wholesale in each command.

Reply:
Changed back now.